internal/v1/handler_blueprints_test.go

```diff
@@ -0,0 +1,98 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/osbuild/image-builder/internal/common"
+	"github.com/osbuild/image-builder/internal/composer"
+	"github.com/stretchr/testify/require"
+
+	"github.com/osbuild/image-builder/internal/tutils"
+)
+
+func TestHandlers_ComposeBlueprint(t *testing.T) {
+	ids := []uuid.UUID{uuid.New(), uuid.New()}
+	idx := 0
+	apiSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if "Bearer" == r.Header.Get("Authorization") {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+		require.Equal(t, "Bearer accesstoken", r.Header.Get("Authorization"))
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		result := composer.ComposeId{
+			Id: ids[idx],
+		}
+		idx += 1
+		err := json.NewEncoder(w).Encode(result)
+		require.NoError(t, err)
+	}))
+	defer apiSrv.Close()
+
+	dbase, err := dbc.NewDB()
+	require.NoError(t, err)
+	srv, tokenSrv := startServerWithCustomDB(t, apiSrv.URL, "", dbase, "../../distributions", "")
+	defer func() {
+		err := srv.Shutdown(context.Background())
+		require.NoError(t, err)
+	}()
+	defer tokenSrv.Close()
+
+	id := uuid.New()
+	versionId := uuid.New()
+
+	uploadOptions := UploadRequest_Options{}
+	err = uploadOptions.FromAWSUploadRequestOptions(AWSUploadRequestOptions{
+		ShareWithAccounts: common.ToPtr([]string{"test-account"}),
+	})
+	require.NoError(t, err)
+	blueprintBody := BlueprintV1{
+		Version:     1,
+		Name:        "blueprint",
+		Description: "desc",
+		Customizations: Customizations{
+			Packages: common.ToPtr([]string{"nginx"}),
+		},
+		Distribution: "centos-8",
+		ImageRequests: []ImageRequest{
+			{
+				Architecture: ImageRequestArchitectureX8664,
+				ImageType:    ImageTypesAws,
+				UploadRequest: UploadRequest{
+					Type:    UploadTypesAws,
+					Options: uploadOptions,
+				},
+			},
+			{
+				Architecture: ImageRequestArchitectureAarch64,
+				ImageType:    ImageTypesAws,
+				UploadRequest: UploadRequest{
+					Type:    UploadTypesAws,
+					Options: uploadOptions,
+				},
+			},
+		},
+	}
+
+	var message []byte
+	message, err = json.Marshal(blueprintBody)
+	err = dbase.InsertBlueprint(id, versionId, "000000", "000000", "blueprint", "desc", message)
+	require.NoError(t, err)
+
+	respStatusCode, body := tutils.PostResponseBody(t, fmt.Sprintf("http://localhost:8086/api/image-builder/v1/experimental/blueprint/%s/compose", id.String()), map[string]string{})
+	require.Equal(t, http.StatusCreated, respStatusCode)
+
+	var result []ComposeResponse
```

Comment:
I was bit surprised that we respond with array of objects each containing ID, feels bit odd and I missed that on review.
Do we want to think about the response format bit more @amirfefer and @lzap, or is it what we want?
Also happy for your opinion @avitova :)

Keeping in draft until we decide that, if we want to change the response format, this PR might be perfect opportunity :)

Reply:
Oh yeah. You might be correct. We might want to use ComposesResponse, right?

Reply:
Well the other option what I considered was to return nothing. This is a blueprint REST endpoint after all.

Are you asking about what to return, or how we return it? Returning slice is definitely not a good practice, we should return a type with some sort of root element just in case we need pagination later. I suggest to add it into our refactoring issue perhaps: https://issues.redhat.com/browse/HMS-3308

Reply:
Yup, added it there, lets not add to the scope of the PR :+1: